### PR TITLE
GA tracking

### PR DESCRIPTION
**Pull request by shaundillon**
This adds GA tracking and a couple of helpers. The pages are currently tracked, but there are currently no events. We should decide what we want to track further.

### CODE REVIEW

File: app/client/components/main.tsx [hunk 1]
```diff
@@ -1,36 +1,44 @@
 import React from "react";
+import { initGA } from "../analytics";
 import palette from "../colours";
 import Footer from "./footer";
 import Header from "./header";
 
-export const Main: React.SFC<{}> = ({ children }) => {
```
Comment (shaundillon) [inline app/client/components/main.tsx]: We should take a look at reducing the amount of SFCs at this high level in the tree, as they'll *always* re-render when their children change, even if they don't need to. `PureComponent` is more performant. SFCs should be saved for leaf nodes.

File: app/client/analytics.ts [hunk 2]
```diff
@@ -1,22 +1,44 @@
...
 })();
+
+interface Event {
+  eventCategory: string;
+  eventAction: string;
+  eventLabel?: string;
+  eventValue?: number;
+}
+
+const trackEvent = ({
```
Comment (shaundillon) [inline app/client/analytics.ts]: This isn't used anywhere yet, but should be for everything other than pageviews

File: app/server/html.ts [hunk 3]
```diff
@@ -49,9 +49,14 @@ const html: (
...
-     ${raven(dsn)}
+      ${raven(dsn)}
+      <script>
+        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
+        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+        })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
+      </script>
       <script src="${src}"></script>
-      <script async src="https://www.googletagmanager.com/gtag/js?id=UA-51507017-5"></script>
```
[the remark this answers is not in the document]
Comment (twrichards) [reply app/server/html.ts]: The Tag Manager approach wasn't populating the view in Google Analytics dashboard - this new approach definitely works though

File: app/client/components/main.tsx (hunk 1 above)
Comment (twrichards) [reply app/client/components/main.tsx]: good suggestion - feel free to change in your responsiveness PR

File: app/server/html.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (svillafe) [reply app/server/html.ts]: In order to GTM to populate correctly GA we need to set it up in GTM console. It is good that this approach works but probably if we would like to be consistent across the Reader Revenue projects we would consider restoring GTM. (Additionally via GTM it is easier to set up other tracking scripts)

[the remark this answers is not in the document]
Comment (twrichards) [reply app/server/html.ts]: thanks. after some discussion we've decided to go with the approach in this PR for the time being. we will be revisiting soon anyway when we look at setting up Google Optimize

**Pull request merged**